components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/JWTUtils.java

```diff
@@ -468,8 +468,12 @@ public static X509Certificate resolveSignerCertificate(IdentityProvider idp) thr
         X509Certificate x509Certificate;
         String tenantDomain = getTenantDomain();
         try {
-            x509Certificate = (X509Certificate) IdentityApplicationManagementUtil
-                    .decodeCertificate(idp.getCertificate());
+            if (IdentityApplicationConstants.RESIDENT_IDP_RESERVED_NAME.equals(idp.getIdentityProviderName())) {
+                x509Certificate = (X509Certificate) OAuth2Util.getCertificate(tenantDomain);
+            } else {
+                x509Certificate = (X509Certificate) IdentityApplicationManagementUtil
+                        .decodeCertificate(idp.getCertificate());
+            }
         } catch (CertificateException e) {
             throw new IdentityOAuth2Exception("Error occurred while decoding public certificate of Identity Provider "
                     + idp.getIdentityProviderName() + " for tenant domain " + tenantDomain, e);
```

Comment:
Suggest to use java string formatters as there are having multiple concatenations.
https://www.w3schools.com/java/ref_string_format.asp

Reply:
This is part of the legacy code. As of now, will refrain from making any other changes.